react-components/src/components/RevealToolbar/RuleBasedOutputsButton.tsx

```diff
@@ -77,18 +99,36 @@ export const RuleBasedOutputsButton = ({
                 marginBottom: '20px'
               }}>
               <Menu.Header>{t('RULESET_SELECT_HEADER', 'Select color overlay')}</Menu.Header>
+              <Menu.Item key="no-rule-selected">
+                <Flex justifyContent="space-between" alignItems="center" gap={8}>
+                  <Flex gap={4} alignItems="center">
+                    <Icon type="ColorPalette" />
+                    {t('RULESET_NO_SELECTION', 'No RuleSet selected')}
+                  </Flex>
+                  <Radio
+                    name={'no-selection'}
+                    value={'no-selection'}
+                    checked={emptyRuleSelected?.isEnabled}
+                    onChange={(_: any, value: string | undefined) => {
+                      onChange(value);
+                    }}
+                  />
+                </Flex>
+              </Menu.Item>
               {ruleInstances?.map((item) => (
-                <Menu.Item key={item.rule.properties.id}>
+                <Menu.Item key={item?.rule?.properties.id}>
                   <Flex justifyContent="space-between" alignItems="center" gap={8}>
                     <Flex gap={4} alignItems="center">
                       <Icon type="ColorPalette" />
-                      {item.rule.properties.name}
+                      {item?.rule?.properties.name}
                     </Flex>
-                    <Switch
-                      checked={item.isEnabled}
-                      id={item.rule.properties.id}
-                      name={item.rule.properties.id}
-                      onChange={onChange}
+                    <Radio
+                      name={item?.rule?.properties.id}
+                      value={item?.rule?.properties.id}
+                      checked={item?.isEnabled}
+                      onChange={(_: any, value: string | undefined) => {
+                        onChange(value);
+                      }}
                     />
```

Comment:
Could it make sense to factor out a common component for each list item? And let the "empty state" use that one as well?

Reply:
https://github.com/cognitedata/reveal/pull/4453/commits/5824d886e02f0d1ee6ecdf3c0738b52e885dd19f